constraintsolver/include/CNSat.h

```diff
@@ -9,14 +9,14 @@
 #define CNSAT_H_
 
 #include "types/Assignment.h"
+#include "engine/AlicaClock.h"
```

Comment:
Why do remove the forward declaration here?

Reply:
Ah, I overlooked the pointer to AlicaClock. In the case of pointers to objects, the class can just be forward declared. Fixed.